### PR TITLE
Add abseil-cpp LTS 20230802.0 to BCR

### PR DESCRIPTION
**Pull request by morambro**

### CODE REVIEW

**Comment by morambro** [comment]
The [20230802.0 release](https://github.com/abseil/abseil-cpp/releases/tag/20230802.0) doesn't have a release archive. What are the options here to avoid the error:
```
BcrValidationResult.FAILED: abseil-cpp@20230802.0 is using an unstable source url: `https://github.com/abseil/abseil-cpp/archive/refs/tags/20230802.0.tar.gz`.
--
  | You should use a release archive URL in the format of `https://github.com/<ORGANIZATION>/<REPO>/releases/download/<version>/<name>.tar.gz` to ensure the archive checksum stability.
  | See https://blog.bazel.build/2023/02/15/github-archive-checksum.html for more context.
```

**Comment by fmeum** [comment]
@morambro Could you file an issue for this upstream? If they usually add release archives, it would be great if they could keep doing that. Skipping the check is only a bandaid.

**Review by fmeum — APPROVED** [review]

**Comment by fmeum** [comment]
We can always add the archive later though, so this can go in now.

File: modules/abseil-cpp/20230802.0/MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+module(
+    name = "abseil-cpp",
+    version = "20230802.0",
+    compatibility_level = 1,
+)
+bazel_dep(name = "rules_cc", version = "0.0.8")
+bazel_dep(name = "platforms", version = "0.0.7")
+bazel_dep(name = "bazel_skylib", version = "1.4.1")
+bazel_dep(name = "googletest", version = "1.14.0", repo_name = "com_google_googletest")
+bazel_dep(name = "google_benchmark", version = "1.8.2", repo_name = "com_github_google_benchmark")
```
Comment (meteorcloudy) [inline modules/abseil-cpp/20230802.0/MODULE.bazel:11]: I was trying to upgrade abseil to fix compatibility with Xcode 15, turns out abseil-cpp introduced a bunch of new dependencies through google_benchmark. 

Checking https://github.com/search?q=repo%3Aabseil%2Fabseil-cpp%20com_github_google_benchmark&type=code, it seems google_benchmark can be specified as a dev dependency?

**Pull request merged**